### PR TITLE
Vendoring Shipwright v0.9.0

### PR DESCRIPTION
**Pull request by HeavyWombat**
# Changes

Update Tekton Pipelines version to `v0.34.1`.

Update Shipwright Build version to `v0.9.0`.

Update internals where needed to match the changed Build API.

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [X] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [X] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,8 +23,8 @@ OUTPUT_HOSTNAME ?= registry.registry.svc.cluster.local:32222
 OUTPUT_NAMESPACE ?= shipwright-io
 
 # Tekton and Shipwright Build Controller versions for CI
-TEKTON_VERSION ?= v0.30.0
-SHIPWRIGHT_VERSION ?= v0.8.0
+TEKTON_VERSION ?= v0.34.1
+SHIPWRIGHT_VERSION ?= nightly-2022-04-07-1649308275
```
Comment (HeavyWombat) [inline Makefile:27]: just as a stand-in at the moment, will be replaced with v0.9.0

File: pkg/shp/flags/buildrun.go [hunk 2]
```diff
@@ -41,9 +44,14 @@ func SanitizeBuildRunSpec(br *buildv1alpha1.BuildRunSpec) {
...
 	}
+	if br.BuildRef != nil {
+		if br.BuildRef.Name == "" && br.BuildRef.APIVersion != nil && *br.BuildRef.APIVersion == "" {
+			br.BuildRef = nil
+		}
+	}
 	if br.ServiceAccount != nil {
 		if (br.ServiceAccount.Name == nil || *br.ServiceAccount.Name == "") &&
-			!br.ServiceAccount.Generate {
+			!*br.ServiceAccount.Generate {
```
Comment (HeavyWombat) [inline pkg/shp/flags/buildrun.go:54]: I am not a particular fan of this notation, if other feel the same, we could do a `== false` or similar.

Comment (HeavyWombat) [reply pkg/shp/flags/buildrun.go]: Anyway, the linter will report the `== false` style, so keeping it like this.

**Review by SaschaSchwarze0 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**